frontend_csv/src/parser/ast_msg.rs

```diff
@@ -0,0 +1,190 @@
+use crate::scanner::token::CSVTokenType;
+use std::fmt::Debug;
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum AstMsgLineType {
+    Msgtype,
+    MsgdataLength,
+    MsgDataBytes,
+    MsgDataTLVInit,
+    Tlvtype,
+    Tlvdata,
+    TlvByteDataWithDot,
+    TvlChainWithDot,
+    // temporary place holder for no match
+    None,
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum AstMsgType {
+    Init,
+    Warning,
+    Error,
+    Ping,
+    Pong,
+    // temporary place holder for no match
+    None,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct AstMsg {
+    pub msg_type: AstMsgType,
+    pub msg_val: String,
+}
+
+pub struct InitMsg {
+    pub raw: Vec<(AstMsgLineType, Vec<String>)>,
+    pub msg_type: AstMsgType,
+    pub msg_val: String,
+    pub msg_name: String,
+    pub tlvs_name: Vec<String>,
+}
+
+pub trait AstMsgTrait {
+    fn get_msg_type(&self) -> AstMsgType;
+    fn get_values(&self) -> String;
+    fn check_valid(&mut self) -> bool;
+}
+
+impl InitMsg {
+    pub fn new(lines: Vec<(AstMsgLineType, Vec<String>)>) -> InitMsg {
+        return InitMsg {
+            raw: lines,
+            msg_type: AstMsgType::Init,
+            msg_val: String::new(),
+            msg_name: String::new(),
+            tlvs_name: Vec::new(),
+        };
+    }
+}
+
+impl AstMsgTrait for InitMsg {
```

Comment:
```suggestion
impl AstMsgTrait for InitMsg {
```

Why we are parsing the specific Init Message? and not build a generic message?